concourse/scripts/run_job.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash -eu
+set -o pipefail
+
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+PIPE='create-bosh-cloudfoundry'
+JOB=$1
+TEMP="/tmp/concourse-run-job.${RANDOM}"
+trap "rm -f $TEMP" ERR
+
+export TARGET_CONCOURSE=deployer
+# shellcheck disable=SC2091
+$("${SCRIPT_DIR}/environment.sh")
+"${SCRIPT_DIR}/fly_sync_and_login.sh"
+FLY="$FLY_CMD -t ${FLY_TARGET}"
+
+# 2 steps prevent fly reading empty input in case step 1 fails
```

Comment:
There are several issues:
1. I initially did pipefail and <get manifest> | remove_passed | upload manifest. The problem is that this starts all 3 processes in parallel. If 1st process fails (e.g. due to network connectivity). Or 2nd process breaks (because of e.g. bug in code or broken yaml on input), the last process will receive empty input - and will upload it. Effectively deleting the pipeline.
2. Then I thought about using named pipes instead of a temp file. However named pipes are blocking. Which means I'd have to do (a) <get manifest> | remove_passed >pipe & (meaning, let it run in the background). And then (b) <upload manifest> <pipe. Which would either get stuck forever if there is no output from (a), or read empty/broken input in case (a) fails (this setup thus makes set -e/pipefail ineffective for capturing/preventing errors).
3. So in the end I chose the temp file (even though I don't like it so much). Because I can do 1st step where I render the mainfest, but if that fails, set -e/pipefail will prevent 2nd step. I have also thought that the pipeline could contain sensitive data (e.g. passwords), but we render pipeline on the disk anyway before uploading, so that data is on disk anyaway.

Nevertheless, I am happy for suggestions how to solve this in a better way. I don't want to trigger specific pipeline rendering scripts again (& then remove passed & upload), as:
* that would make the solution very pipeline specific
* would overwrite any changes user did by hand to the pipeline